### PR TITLE
samples: publish with ordering keys

### PR DESCRIPTION
**Pull request by anguillanneuf**
Add samples for publish with ordering keys. The test can't be run yet because the feature isn't released:
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: samples/snippets/subscriber.py [hunk 1]
```diff
@@ -160,6 +160,28 @@ def create_push_subscription(project_id, topic_id, subscription_id, endpoint):
...
+    # [START pubsub_enable_subscription_ordering]
+    from google.cloud import pubsub_v1
+
+    # TODO(developer): Choose an existing topic.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+    # subscription_id = "your-subscription-id"
+
+    subscriber = pubsub_v1.SubscriberClient()
+    topic_path = subscriber.topic_path(project_id, topic_id)
```
Comment (plamut) [inline samples/snippets/subscriber.py:174]: `topic_path()` does not exist anymore on subscriber client, had to fix quite a few of these in #158. :)

Comment (anguillanneuf) [reply samples/snippets/subscriber.py:174]: That is a massive PR! Thank you for the heads-up! I will aim to merge this one after that one and the changes are in.

File: samples/snippets/publisher.py [hunk 2]
```diff
@@ -273,6 +273,82 @@ def publish_messages_with_retry_settings(project_id, topic_id):
...
+    """Publishes messages with ordering keys."""
+    # [START pubsub_publish_with_ordering_keys]
+    from google.cloud import pubsub_v1
+
+    # TODO(developer): Choose an existing topic.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+
+    publisher_options = pubsub_v1.types.PublisherOptions(
+        enable_message_ordering=True
```
Comment (kamalaboulhosn) [inline samples/snippets/publisher.py:286]: Can we also add use of a regional endpoint, e.g., us-east1-pubsub.googleapis.com:443? Using a regional endpoint for ordering is a best practice.

File: samples/snippets/publisher.py [hunk 3]
```diff
@@ -273,6 +273,82 @@ def publish_messages_with_retry_settings(project_id, topic_id):
...
+    """Resume publishing messages with ordering keys when unrecoverable errors occur."""
+    # [START pubsub_resume_publish_with_ordering_keys]
+    from google.cloud import pubsub_v1
+
+    # TODO(developer): Choose an existing topic.
+    # project_id = "your-project-id"
+    # topic_id = "your-topic-id"
+
+    publisher_options = pubsub_v1.types.PublisherOptions(
+        enable_message_ordering=True
```
Comment (kamalaboulhosn) [inline samples/snippets/publisher.py:322]: Likewise here for the regional endpoint.

File: samples/snippets/publisher.py (hunk 2 above)
Comment (anguillanneuf) [reply samples/snippets/publisher.py:286]: Done.

File: samples/snippets/publisher.py (hunk 3 above)
Comment (anguillanneuf) [reply samples/snippets/publisher.py:328]: Done.

**Comment by plamut** [comment]
@anguillanneuf How urgent it is to get this in / publish it? 

If possible, we would like to avoid making a yet another "last" Python 2 release and potentially extra work on the PR transitioning the library to the microgenerator.

**Comment by anguillanneuf** [comment]
@plamut Thanks for checking in with me. No need for another Python 2 release. I have not yet "published" the sample on Cloud site. Currently on the Java sample is.

**Pull request merged**